### DISCUSSION TITLE
Using a certificate instead of a client secret

### DISCUSSION THREAD

**Discussion by mikez** [Azure-Samples/ms-identity-python-webapp#147] [Q&A] [answered]
In the Entra admin center it says "For a higher level of assurance, we recommend using a certificate (instead of a client secret) as a credential".

Is this something that the "identity" library or "msal" can help with?
Here's a reference:
https://learn.microsoft.com/en-us/entra/identity-platform/certificate-credentials

**Comment by rayluo** [comment, marked as answer]
By default, [this web app sample demonstrates reading `client_credential` from an ENV VAR](https://github.com/Azure-Samples/ms-identity-python-webapp/blob/0.9.0/app.py#L15) which is typically a client secret string.
However, if you somehow organize your certificate information in the format described by [`msal.ConfidentialClientApplication`'s `client_credential` parameter](https://msal-python.readthedocs.io/en/latest/#msal.ClientApplication.params.client_id), this sample is expected to use that certificate. You can give it a try.
